draft-ietf-quic-load-balancers.md

```diff
@@ -324,6 +308,78 @@ the packet. The number of servers is usually consistent on the time scale of a
 QUIC connection handshake. Another might simply hash the address/port 4-tuple.
 See also {{version-invariance}}.
 
+Note that any information used by the Fallback Algorithm might not be repeated
+from one QUIC packet to another. The 4-tuple is assumed to remain stable during
+the handshake, but the server can change the destination connection ID after the
+client's first flight.
+
+Therefore, if the fallback algorithm uses information other than the 4-tuple in
+its routing decision, the load balancer SHOULD store the routing decision for
+that 4-tuple in order to route future packets unless doing so would exceed
+resource limits. It can free these records when the 4-tuple uses a routable CID.
+This state allows the load balancer to successfully route other packets from
+the client's first flight, as well as subsequent flights if the server provides
+an unroutable CID for the reasons described in {{config-failover}}.
+
+Even if the fallback algorithm solely uses the 4-tuple, it MAY store the
+algorithm result in case changes to the server pool change fallback algorithm
+output.
+
+The load balancer MAY also store the routing decision indexed by DCID. This
+is not robust to new server-selected DCIDs, but can increase robustness when
+a connection to a server generating unroutable CIDs experiences a NAT rebinding.
+An entry can be freed when the corresponding 4-tuple uses a different CID,
+whether routable or not.
+
+### Load Balancer Forwarding {#load-balancer-forwarding}
+
+Packets with routable CIDs are, of course, routed in accordance with the
+decoded server ID.
+
+For unroutable CIDs, load balancers execute the following steps in order
+until one results in a routing decision.
+
+1. If the load balancer has a table of routing decisions by DCID, and the
+DCID is present in it, route the packet accordingly.
+1. If the load balancer has a table of routing decisions by 4-tuple, and
+the DCID is present in it, route the packet accordingly.
+1. Use the fallback algorithm to make a routing decision and, if applicable, record the results
+in the tables indexed by 4-tuple and/or DCID.
+
```

Comment:
Yes. Note that it is much easier to discuss Fallback Algorithms *after* we have defined the default behavior. Consider changing the order of these sections.